ui/pages/confirm-transaction-base/confirm-transaction-base.test.js

```diff
@@ -33,6 +33,7 @@ const baseStore = {
     unapprovedTxs: {
       1: {
         id: 1,
+        metamaskNetworkId: '5',
```

Comment:
For some reason these lines were necessary to add in order to make these tests pass. I am not quite sure why, but I also don't know how these tests passed originally anyway considering these transaction objects aren't complete. I can look into this next week.

Reply:
Curious. I found that removing these two lines and adding the property `chainId: CHAIN_IDS.GOERLI,` here also gets them passing.

I guess there was some fallback case for when `chainId` wasn't set that was getting triggered here previously

Reply:
Ah, I think I found the problem. There are still references to `network` around. I found one in `ui/pages/confirm-signature-request/index.js`

Reply:
This specific problem is probably caused by the `network` reference in `incomingTxListSelector`

Reply:
Oh, found one in `confirm-transaction-base.container.js`. That's even closer to this test.

Reply:
Found another in `ui.components/app/transaction-alerts/transaction-alerts.stories.js`. I'm finished my `network` audit, that's all that I found.

Reply:
Thanks! Though I think there is still one left, `incompingTxListSelector`, in `ui/selectors/transactions.js`

Reply:
Also, I took another look at this test, and the reason why it's failing is due to a change in `deprecatedGetCurrentNetworkId` in `ui/selectors/selectors.js`, but also, my feeling is (still) that the test was not written correctly in the first place.

Previously, `deprecatedGetCurrentNetworkId` returned `state.metamask.network`. Also, in state object that the test sets up, three things were not set: 1) `network`, 2) `metamaskNetworkId` in the unapproved transaction object, and 3) `chainId` in the unapproved transaction object. This meant that when the `unconfirmedTransactionsHashSelector` selector in `ui/selectors/confirm-transaction.js` looked for transactions that matched the current network ID (obtained via `deprecatedGetCurrentNetworkId`), it called `transactionMatchesNetwork({ ..., metamaskNetworkId: undefined, chainId: undefined }, undefined, undefined)`. `undefined` matches `undefined`, so all of the unapproved transactions defined in the test matched.

Now `deprecatedGetCurrentNetworkId` doesn't return `state.metamask.network`, it returns `state.metamask.networkId ?? 'loading'`. That means it can never be undefined. However, the state object in the test is still not correct, so `unconfirmedTransactionsHashSelector` now calls `transactionMatchesNetwork({ ..., metamaskNetworkId: undefined, chainId: undefined }, 'loading', undefined)`. `'loading'` no longer matches `undefined`, so none of the unapproved transactions defined in the test matched.

To fix this, `networkId` should be set to something reasonable, and the unapproved transaction should have at least a `metamaskNetworkId` that matches (`chainId` would make sense too, I suppose).

Reply:
Thanks, that makes sense

Reply:
> Thanks! Though I think there is still one left, `incompingTxListSelector`, in `ui/selectors/transactions.js`

Updated in [ce1a1d0](https://github.com/MetaMask/metamask-extension/pull/17556/commits/ce1a1d0c29837a8038dcb4ea5f7850c0a4fba1ed).